fault-tolerance/fault-tolerance/src/main/java/io/helidon/faulttolerance/FtGenerated.java

```diff
@@ -0,0 +1,237 @@
+/*
+ * Copyright (c) 2024, 2025 Oracle and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.helidon.faulttolerance;
+
+import io.helidon.common.Functions;
+import io.helidon.common.UncheckedException;
+import io.helidon.service.registry.Interception;
+import io.helidon.service.registry.InterceptionContext;
+
+/**
+ * Types used only in generated code (and of course when reading the generated code).
+ */
+public final class FtGenerated {
```

Comment:
Do we have a precedent for a *Generated API ?

I don't like the use of the the word "generated". This is still a public API and it should be usable manually without a code generator.

IIUC it starts with `Ft` to be logically grouped with the `Ft` container class. How about `FtSupport` instead ?

Reply:
we have `io.helidon.builder.api.GeneratedBuilder`
not really consistent. Will think about this.